core/src/modules/configLoaders/index.ts

```diff
@@ -118,6 +118,8 @@ export async function processConfigObjects(
   );
   errors.push(...validationErrors);
 
+  if (errors.length > 0) return { seenIds, errors };
```

Comment:
@evantahler I put this here, but it seems like there may be a better way to go about all this.

With this in place, let's say I have a duplicate `id`, but I also have other issues. When I `validate` the first time, I see:

```
🦘 Grouparoo: validate

info: validating 3 objects...
error: [ config ] Duplicate ID values found: data_warehouse
error: ❌ Validation failed - 1 validation error
```

So I fix my `data_warehouse` duplicate issue and run `validate` again:

```
🦘 Grouparoo: validate

info: validating 3 objects...
notice: [ app ] testing app data_warehouse (data_warehouse) threw error: role "..." does not exist
error: [ config ] error with app `data_warehouse` (data_warehouse): Error: error testing app data_warehouse (data_warehouse) - role "..." does not exist
error: [ config ] error with property `data_warehouse` (data_warehouse_prop): Error: cannot find Source with id "..."
error: [ config ] error with source `gsheet` (gsheet): Error: cannot find App with id "..."
error: ❌ Validation failed - 3 validation errors
```

Maybe that's okay. Or unavoidable. What do you think?

Reply:
I think it's OK for now at least!

My main concern was that you, perhaps, made 2 sources called "users-table" (from different apps).  If they were both valid (had all the info they needed) you would replace one of them with the other.  I think that you need to eject quickly if you see that, before any of the `grouparoo apply` logic could be happening